R/MsBackendDataFrame.R

```diff
@@ -52,12 +55,16 @@ setMethod("show", "MsBackendDataFrame", function(object) {
 #'
 #' @rdname MsBackend
 setMethod("backendInitialize", signature = "MsBackendDataFrame",
-          function(object, data, ...) {
+          function(object, data, peaksVariables = c("mz", "intensity"), ...) {
               if (missing(data)) data <- DataFrame()
               if (is.data.frame(data))
                   data <- DataFrame(data)
               if (!is(data, "DataFrame"))
                   stop("'data' has to be a 'DataFrame'")
+              peaksVariables <- intersect(peaksVariables, colnames(data))
+              if (sum(c("mz", "intensity") %in% peaksVariables) == 1L)
```

Comment:
If I understand this check, it will throw an error of only one of "mz" or "intensity" are in the peaksVariables. If none are, this wouldn't trigger the error. Why not 

```
sum(c("mz", "intensity") %in% peaksVariables) != 2L
```

Reply:
good catch! thanks, changed that.

Reply:
actually, after checking again: we were supporting the following situations:

- user provides m/z and intensity
- user does not provide neither m/z and intensity (in which case 0-length spectra are available, but the `Spectra` is still considered valid).

An error is only thrown if either m/z but not intensity, or intensity but not m/z is provided. from a data point of view that does not make sense, you can't have one of the two. you could either have both or none.

Open to discuss @lgatto when you think that's not a good idea.

Reply:
OK, fine by me. May be just clarify with a comment?

Reply:
jepp, will add that and also check that I mention it in documentation

Reply:
added as a comment and docu.